lib/src/components/breadcrumb.dart

```
@@ -0,0 +1,297 @@
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return ShadBreadcrumbItem(
+      isCurrentPage: true,
+      child: child,
+    );
+  }
+}
+
+/// {@template ShadBreadcrumbSeparator}
+/// A separator widget used between breadcrumb items.
+///
+/// This widget provides a visual separator between breadcrumb items.
+/// By default, it displays a chevron right icon, but can be customized
+/// with any widget.
+/// {@endtemplate}
+class ShadBreadcrumbSeparator extends StatelessWidget {
+  /// {@macro ShadBreadcrumbSeparator}
+  const ShadBreadcrumbSeparator({
+    super.key,
+    this.child,
+  });
+
+  /// The widget to display as the separator.
+  /// If null, uses the default chevron right icon.
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = ShadTheme.of(context);
+    final breadcrumbTheme = theme.breadcrumbTheme;
+
+    return child ??
+        breadcrumbTheme.separator ??
+        Icon(
+          LucideIcons.chevronRight,
+          size: 16,
+          color: theme.colorScheme.mutedForeground,
+        );
+  }
+}
+
+/// {@template ShadBreadcrumbEllipsis}
+/// An ellipsis indicator for breadcrumb navigation.
+///
+/// This widget is used to indicate that there are more breadcrumb items
+/// that are not currently visible, typically in a collapsed state.
+/// It's often used with dropdown menus to show hidden breadcrumb levels.
+/// {@endtemplate}
+class ShadBreadcrumbEllipsis extends StatelessWidget {
+  /// {@macro ShadBreadcrumbEllipsis}
+  const ShadBreadcrumbEllipsis({
+    super.key,
+    this.child,
+  });
+
+  /// The widget to display as the ellipsis.
+  /// If null, uses the default more horizontal icon.
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = ShadTheme.of(context);
+    final breadcrumbTheme = theme.breadcrumbTheme;
+
+    return SizedBox(
+      width: 36,
+      height: 36,
```

These hardcoded values are not good, should be customisable both from the widget and theme levels

---

@nank1ro Modified it to, in order, take the size from the widget, then the theme and a fallback to hardcoded values (to prevent a layout painting error)

---

Perfect! It's correct to fallback to a default value